### ISSUE TITLE
Wrong return value when returing array from function

### ISSUE THREAD

**Issue by ggVGc** [bryanjos/elixirscript#195]
The following code prints "2" when it should print "[1,2]"
```
  def foo(x) do
    [x.a, x.b]
  end

  dict = %{
    :a=>1,
    :b => 2
  }
  :console.log(foo dict)
```
If the function is changed to the following, it behaves correctly.

```
  def foo(x) do
    ret = [x.a, x.b]
    ret
  end
```

**Comment by bryanjos** [comment]
Thanks I'll take a look

**Comment by bryanjos** [comment]
resolved in master f677d2ffdc5349b7a24308fdf89fad4116448b3a
